### PR TITLE
Project functionality is moved into Lib

### PR DESCRIPTION
**Pull request by andped10**
Using the functionality in the Lib

### CODE REVIEW

File: EasyReflectometryApp/Logic/Proxies/Project.py [hunk 1]
```diff
@@ -112,213 +92,38 @@ def currentProjectPath(self, new_path):
...
+        self._project.create_project_dir()
+        self._project.default_model()
+        self._relayProjectChange()
+        self.saveProject()
 
-    @Slot()
-    def saveProject(self):
-        self._saveProject()
+    def _relayProjectChange(self):
+        self.parent._model_proxy._model = self._project._models
```
Comment (rozyczko) [inline EasyReflectometryApp/Logic/Proxies/Project.py:101]: This makes `ProjectProxy` aware of other proxies. This should be avoided.
How about telling the main proxy to deal with the assignment?
E.g.
```python
self.parent.assign_model(self._project._models)
```
and the main proxy would just forward this to whichever proxy needs it, say
```python
def assign_model(self, model):
    self._model_proxy._model = model
```

File: EasyReflectometryApp/Logic/Proxies/Project.py [hunk 2]
```diff
@@ -112,213 +92,38 @@ def currentProjectPath(self, new_path):
...
     @Slot()
     def loadProject(self):
-        self._loadProject()
+        self._project.load_project_json()
+        self._relayProjectChange()
+
+    @Slot()
+    def saveProject(self):
+        self._project.save_project_json()
         self.parent._state_proxy.stateChanged.emit(False)
```
Comment (rozyczko) [inline EasyReflectometryApp/Logic/Proxies/Project.py:122]: This is an old code, but please also consider not calling `_state_proxy` directly.
Could be done as a separate maintenance fix.

**Review by rozyczko — APPROVED** [review]
Looks good.
There is a general issue of not calling outside proxies, but I can see it is very widespread.
This probably should be addressed/fixed at some point, but we may start with not adding to this mess in the new code :)

File: EasyReflectometryApp/Logic/Proxies/Project.py (hunk 1 above)
Comment (andped10) [reply EasyReflectometryApp/Logic/Proxies/Project.py:101]: I expect that cases like this should disappear if we have `project` as the main data structure that all proxies know.  What I mean is that all the different proxies will be able to manipulate the `project` but each of the proxies  should be focused on a given part of `project`.

I will try to use that code structure in the `src_qt6` and leave the code as is for the "old" qt5 GUI.

**Pull request merged**